packages/react/popover/src/Popover.tsx

```diff
@@ -167,14 +170,106 @@ const PopoverContent = React.forwardRef((props, forwardedRef) => {
+     */
+    portalled?: boolean;
+  }
+>;
+
+type PopoverContentTypePrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof PopoverContentImpl>,
+  PopoverContentTypeOwnProps
+>;
+
+const PopoverContentModal = React.forwardRef((props, forwardedRef) => {
+  const { portalled = true, ...contentModalProps } = props;
+  const context = usePopoverContext(CONTENT_NAME);
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, contentRef);
+
+  // aria-hide everything except the content (better supported equivalent to setting aria-modal)
+  React.useEffect(() => {
+    const content = contentRef.current;
+    if (content) return hideOthers(content);
+  }, []);
+
+  const PortalWrapper = portalled ? Portal : React.Fragment;
+
+  return (
+    <PortalWrapper>
+      <RemoveScroll>
+        <PopoverContentImpl
+          {...contentModalProps}
+          ref={composedRefs}
+          // we make sure we're not trapping once it's been closed
+          // (closed !== unmounted when animating out)
+          trapFocus={context.open}
+          disableOutsidePointerEvents
+          // When focus is trapped, a `focusout` event may still happen.
+          // We make sure we don't trigger our `onDismiss` in such case.
+          onFocusOutside={composeEventHandlers(props.onFocusOutside, (event) =>
+            event.preventDefault()
+          )}
+        />
+      </RemoveScroll>
+    </PortalWrapper>
+  );
+}) as PopoverContentTypePrimitive;
+
+const PopoverContentNonModal = React.forwardRef((props, forwardedRef) => {
+  const { portalled = true, ...contentNonModalProps } = props;
+  const context = usePopoverContext(CONTENT_NAME);
+  const isPointerDownOutsideRef = React.useRef(false);
+
+  const PortalWrapper = portalled ? Portal : React.Fragment;
+
+  return (
+    <PortalWrapper>
       <PopoverContentImpl
-        data-state={context.open ? 'open' : 'closed'}
-        {...contentProps}
+        {...contentNonModalProps}
         ref={forwardedRef}
+        trapFocus={false}
+        onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+          if (isPointerDownOutsideRef.current) event.preventDefault();
+        })}
+        disableOutsidePointerEvents={false}
+        onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
+          isPointerDownOutsideRef.current = false;
+        })}
+        onPointerDownOutside={composeEventHandlers(props.onPointerDownOutside, (event) => {
+          const { originalEvent } = event.detail;
+          const isLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === false;
+          isPointerDownOutsideRef.current = isLeftClick;
+        })}
+        onInteractOutside={composeEventHandlers(props.onInteractOutside, (event) => {
+          // prevent dismissing when clicking the trigger
+          // as it's already setup to close, otherwise it would close and immediately open.
```

Comment:
nit: we didn't update the comment either in `Dialog` we should

```suggestion
          // prevent dismissing when clicking the trigger
          // (or focusing it also as some platforms will focus on pointer down)
          // as it's already setup to close, otherwise it would close and immediately open.
```

Reply:
Good spot, thanks. I went with this in the end https://github.com/radix-ui/primitives/pull/751/commits/3d92dfb1872a4fb82e7706b406b4bf2f2e9180ba

Let me know if that works for you.

Reply:
Love it.